datarobot_batch_scoring/api_response_handlers/pred_api_v10.py

```diff
@@ -0,0 +1,76 @@
+import operator
+import json
+
+
+def format_data(result, batch, **opts):
+    pred_name = opts.get('pred_name')
+    keep_cols = opts.get('keep_cols')
+    skip_row_id = opts.get('skip_row_id')
+    fast_mode = opts.get('fast_mode')
+    input_delimiter = opts.get('delimiter')
+
+    single_row = result[0]
+    fields = sorted(record['label']
+                    for record in single_row['predictionValues'])
+
+    if pred_name is not None:
+        out_fields = ['row_id', pred_name]
+        # batch scoring returns only positive class for binary tasks if
+        # user specified pred_name option, so we eliminate negative
+        # result
+        if len(fields) == 2 and single_row['prediction'] in fields:
+            fields = [fields[-1]]
+    else:
+        out_fields = ['row_id'] + fields
+
+    pred = [[record['rowId'] + batch.id] + [
+        pred_vals['value'] for pred_vals in
+        sorted(record['predictionValues'],
+               key=operator.itemgetter('label'))
+        if pred_vals['label'] in fields]
+            for record in sorted(result,
+                                 key=operator.itemgetter('rowId'))]
```

Comment:
Still really hard to read.

Reply:
Yeah, I know. I'd like to limit the scope of this PR to just moving code around. I'll add some unit tests to this code area in https://github.com/datarobot/batch-scoring/pull/105 so it's going to be easier to refactor this afterwards.